editor/src/components/canvas/canvas-strategies/strategies/set-border-radius-strategy.tsx

```diff
@@ -0,0 +1,470 @@
+  const element = MetadataUtils.findElementByElementPath(
+    canvasState.startingMetadata,
+    selectedElement,
+  )
+  if (element == null) {
+    return null
+  }
+
+  const borderRadius = borderRadiusFromElement(element)
+  if (borderRadius == null) {
+    return null
+  }
+
+  const elementSize = sizeFromElement(element)
+  const borderRadiusAdjustData = borderRadiusAdjustDataFromInteractionSession(interactionSession)
+
+  const { commands, updatedBorderRadius } = setBorderRadiusStrategyRunResult(
+    borderRadius,
+    borderRadiusAdjustData,
+    elementSize,
+  )
+
+  const mode: BorderRadiusAdjustMode =
+    borderRadius.mode === 'individual' || borderRadiusAdjustData?.modifiers.cmd === true
+      ? 'individual'
+      : 'all'
+
+  return {
+    id: SetBorderRadiusStrategyId,
+    name: 'Set border radius',
+    fitness: 1,
+    controlsToRender: [
+      controlWithProps({
+        control: BorderRadiusControl,
+        props: {
+          mode: mode,
+          selectedElement: selectedElement,
+          elementSize: elementSize,
+          borderRadius: updatedBorderRadius,
+          showIndicatorOnCorner: borderRadiusAdjustData?.corner ?? null,
+        },
+        key: 'border-radius-handle',
+        show: 'visible-except-when-other-strategy-is-active',
+      }),
+    ],
+    apply: () =>
+      strategyApplicationResult([
+        commands(selectedElement),
+        setElementsToRerenderCommand(selectedElements),
+      ]),
+  }
+}
+
+interface BorderRadiusAdjustData {
+  drag: CanvasVector
+  dragStart: CanvasPoint
+  corner: BorderRadiusCorner
+  modifiers: Modifiers
+}
+
+function borderRadiusAdjustDataFromInteractionSession(
+  interactionSession: InteractionSession | null,
+): BorderRadiusAdjustData | null {
+  if (
+    interactionSession == null ||
+    interactionSession.interactionData.type !== 'DRAG' ||
+    interactionSession.activeControl.type !== 'BORDER_RADIUS_RESIZE_HANDLE'
+  ) {
+    return null
+  }
+  return {
+    drag: interactionSession.interactionData.drag ?? canvasVector({ x: 0, y: 0 }),
+    dragStart: interactionSession.interactionData.dragStart,
+    corner: interactionSession.activeControl.corner,
+    modifiers: interactionSession.interactionData.modifiers,
+  }
+}
+
+function deltaFromDrag(drag: CanvasVector, corner: BorderRadiusCorner): number {
+  switch (corner) {
+    case 'tl':
+      return Math.max(drag.x, drag.y)
+    case 'tr':
+      return Math.max(-drag.x, drag.y)
+    case 'bl':
+      return Math.max(drag.x, -drag.y)
+    case 'br':
+      return Math.max(-drag.x, -drag.y)
+    default:
+      assertNever(corner)
+  }
+}
+
+interface BorderRadiusData<T> {
+  mode: BorderRadiusAdjustMode
+  borderRadius: BorderRadiusSides<T>
+}
+
+function borderRadiusFromElement(
+  element: ElementInstanceMetadata,
+): BorderRadiusData<CSSNumberWithRenderedValue> | null {
+  if (element == null || isLeft(element.element) || !isJSXElement(element.element.value)) {
```

Comment:
Could do a `foldEither` somewhere around here so that you don't have to handle drilling into the `Either` by hand.